libpf/pfelf/pfelf.go

```diff
@@ -165,6 +165,23 @@ func GetBuildID(elfFile *elf.File) (string, error) {
 	return getBuildIDFromNotes(sectionData)
 }
 
+// getGoBuildIDFromNotes returns the Go build ID from an ELF notes section data.
+func getGoBuildIDFromNotes(notes []byte) (string, error) {
+	// 0x4 is the "Go Build ID" type. Not sure where this is standardized.
+	buildID, found, err := getNoteString(notes, "Go", 0x4)
+	if err != nil {
+		return "", fmt.Errorf("could not determine BuildID: %v", err)
+	}
+	//nolint:lll
+	// When building Go binaries, Bazel explicitly sets their build ID to "redacted"
+	// see https://github.com/bazelbuild/rules_go/blob/199d8e4827f87d382a85febd0148c1b42fa949cc/go/private/actions/link.bzl#L174.
+	// In that case, we don't want to associate the build ID with the binary in the mapping.
```

Comment:
This comment seems missleading, as the Go Build ID `redacted` does set `found` to true and this check just passes on.
For this special case handling, `buildID` needs to be checked against `redacted`.

Reply:
Indeed, this is a left over from a previous version, I removed it.